src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs

```diff
@@ -146,50 +133,71 @@ public ReadOnlyBuffer(IEnumerable<Memory<T>> buffers)
+        public ReadOnlyBuffer<T> Slice(long offset, long length)
         {
-            var begin = Seek(BufferStart, BufferEnd, start, false);
+            var begin = Seek(BufferStart, BufferEnd, offset, false);
             var end = Seek(begin, BufferEnd, length, false);
             return new ReadOnlyBuffer<T>(begin, end);
         }
 
         /// <summary>
         /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', ending at 'end' (inclusive).
         /// </summary>
-        /// <param name="start">The index at which to begin this slice.</param>
+        /// <param name="offset">The index at which to begin this slice.</param>
         /// <param name="end">The end (inclusive) of the slice</param>
-        public ReadOnlyBuffer<T> Slice(long start, Position end)
+        public ReadOnlyBuffer<T> Slice(long offset, Position end)
         {
             BoundsCheck(BufferEnd, end);
-            var begin = Seek(BufferStart, end, start);
+            var begin = Seek(BufferStart, end, offset);
             return new ReadOnlyBuffer<T>(begin, end);
         }
 
         /// <summary>
-        /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', ending at 'end' (inclusive).
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', and is at most length bytes
         /// </summary>
         /// <param name="start">The starting (inclusive) <see cref="Position"/> at which to begin this slice.</param>
-        /// <param name="end">The ending (inclusive) <see cref="Position"/> of the slice</param>
-        public ReadOnlyBuffer<T> Slice(Position start, Position end)
+        /// <param name="length">The length of the slice</param>
+        public ReadOnlyBuffer<T> Slice(Position start, long length)
         {
-            BoundsCheck(BufferEnd, end);
-            BoundsCheck(end, start);
+            BoundsCheck(BufferEnd, start);
+
+            var end = Seek(start, BufferEnd, length, false);
 
             return new ReadOnlyBuffer<T>(start, end);
         }
 
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', and is at most length bytes
+        /// </summary>
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="length">The length of the slice</param>
+        public ReadOnlyBuffer<T> Slice(int offset, int length) => Slice((long)offset, length);
+
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', ending at 'end' (inclusive).
+        /// </summary>
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="end">The end (inclusive) of the slice</param>
+        public ReadOnlyBuffer<T> Slice(int offset, Position end) => Slice((long)offset, end);
```

Comment:
I think this defeats the purpose of the int overload.

Reply:
I was adding it to get surface to the place we want.

Reply:
Fixed.